Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionTabCocoa.mm

```diff
@@ -180,12 +198,28 @@
     return [m_delegate isPinnedForWebExtensionContext:m_extensionContext->wrapper()];
 }
 
-void WebExtensionTab::toggleReaderMode()
+bool WebExtensionTab::isPrivate() const
+{
+    if (!isValid())
+        return false;
+
+    auto window = this->window();
+    return window ? window->isPrivate() : false;
+}
+
+void WebExtensionTab::toggleReaderMode(CompletionHandler<void(Error)>&& completionHandler)
 {
     if (!isValid() || !m_respondsToToggleReaderMode)
         return;
 
-    [m_delegate toggleReaderModeForWebExtensionContext:m_extensionContext->wrapper()];
+    [m_delegate toggleReaderModeForWebExtensionContext:m_extensionContext->wrapper() completionHandler:^(NSError *error) {
+        if (error) {
+            completionHandler(error.localizedDescription);
+            return;
+        }
+
+        completionHandler(std::nullopt);
```

Comment:
Nit: `completionHandler(error ? error.localizedDescription : std::nullopt);`

Reply:
Was going to log the error above, so I'll add that now.